mailpoet/lib/EmailEditor/Integrations/MailPoet/MailPoetCssInliner.php

```diff
@@ -0,0 +1,25 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\EmailEditor\Integrations\MailPoet;
+
+use MailPoet\EmailEditor\Engine\Renderer\Css_Inliner;
+use MailPoetVendor\Pelago\Emogrifier\CssInliner;
+
+class MailPoetCssInliner implements Css_Inliner {
+  private CssInliner $inliner;
+
+  public function from_html(string $unprocessed_html): self {// phpcs:ignore PSR1.Methods.CamelCapsMethodName.NotCamelCaps
```

Comment:
I can see in this implementation a possible danger when a method `inline_css` or `render` can be used before calling `from_html`. In that case, there is a possible calling method on `null`. In my opinion, we should avoid it. What do you think?

Reply:
Good point, what do you suggest the action should be if the method `render` is called but the `$inliner` is null?

Reply:
I see two approaches here.
1. Throwing a logical exception when the property is not set yet
2. Using a constructor, which uses CssInliner::from_html(). But this could complicate using DI.

I would prefer throwing an exception. What do you think?